code/model/modelread.cpp

```diff
@@ -1953,44 +1955,44 @@ int read_model_file(polymodel * pm, const char *filename, int n_subsystems, mode
 					if (length > 0)
 					{
 						auto base_length = strlen("$glow_texture=");
-						Assert(strstr( (const char *)&props, "$glow_texture=") != NULL);
-						Assert(length > base_length);
-						char *glow_texture_name = props + base_length;
-						
-						if (glow_texture_name[0] == '$')
-							glow_texture_name++;
+						char *glow_texture_start = strstr(props, "$glow_texture=");
+						if ( (length > base_length) && (glow_texture_start != nullptr) && (strlen(glow_texture_start + base_length) > 0) ) {
```

Comment:
I think the `(length > base_length)` check is redundant, given the other conditions.

Reply:
It is redundant; I just carried it over from the old code because I wasn't thinking straight.